File: day2p2.py
from sys import stdin
from collections import Counter

validCount = 0
for line in stdin.readlines():
    rng, target, pwd = line.split(' ')
    minIndex, maxIndex = map(int, rng.split('-'))
    char = target.replace(':', '')
    if (pwd[minIndex-1] == char) ^ (pwd[maxIndex-1] == char):
        validCount += 1
print(f'Valid passwords: {validCount}')

File: day2p2.rs
use std::fs::File;
use std::io::{BufRead, BufReader};


fn main() {
    let reader = BufReader::new(File::open("input/input2.txt").unwrap());
    let mut valid_count = 0;
    for ln in reader.lines() {
        let line = ln.unwrap();
        let arr = line.trim().split(" ").collect::<Vec<&str>>();
        // Get indices, required charater, and password from line
        let indices = arr[0].split("-").map(|s| s.parse::<usize>().unwrap()).collect::<Vec<usize>>();
        let req = arr[1].replace(":", "");
        let pwd = arr[2].split("").collect::<Vec<&str>>();
        if pwd[indices[0]] == req && pwd[indices[1]] == req {
            valid_count += 1;
        }
    }
    println!("{} valid passwords", valid_count);
}